framework/console/Application.php

```diff
@@ -180,8 +181,156 @@ public function runAction($route, $params = [])
             $res = parent::runAction($route, $params);
             return is_object($res) ? $res : (int)$res;
         } catch (InvalidRouteException $e) {
-            throw new Exception("Unknown command \"$route\".", 0, $e);
+            $message = "Unknown command \"$route\"";
+            if($alternatives = $this->findAlternatives($route)){
+                $message .= count($alternatives) > 1 ?  "\nDid you mean one of these?" : "\nDid you mean this?";
+                $message .= "\n    " . implode("\n    ", $alternatives);
+            }
+            throw new Exception($message, 0, $e);
+        }
+    }
+
+    /**
+     *
+     * @param string $route
+     * @return array
+     */
+    public function findAlternatives($route)
+    {
+        $commands = $this->getCommands();
+        $threshold = 1e3;
+        $alternatives = [];
+
+        $commandParts = [];
+        foreach ($commands as $command) {
+            $commandParts[$command] = explode('/', $command);
+        }
+
+        foreach (explode('/', $route) as $i => $subname) {
+            foreach ($commandParts as $command => $parts) {
+                $exists = isset($alternatives[$command]);
+                if (!isset($parts[$i]) && $exists) {
+                    $alternatives[$command] += $threshold;
+                    continue;
+                } elseif (!isset($parts[$i])) {
+                    continue;
+                }
+
+                $lev = levenshtein($subname, $parts[$i]);
+                if ($lev <= strlen($subname) / 3 || '' !== $subname && false !== strpos($parts[$i], $subname)) {
+                    $alternatives[$command] = $exists ? $alternatives[$command] + $lev : $lev;
+                } elseif ($exists) {
+                    $alternatives[$command] += $threshold;
+                }
+            }
+        }
+
+        foreach ($commands as $command) {
+            $lev = levenshtein($route, $command);
+            if ($lev <= strlen($route) / 3 || strpos($command, $route) !== false) {
+                $alternatives[$command] = isset($alternatives[$command]) ? $alternatives[$command] - $lev : $lev;
+            }
+        }
+        
+        $alternatives = array_filter($alternatives, function ($lev) use ($threshold) { 
+            return $lev < 2 * $threshold;
```

Comment:
Copied from Sympony https://github.com/symfony/console/blob/master/Application.php#L971.
I am also not understand what actually is :grin: